sockeye/encoder.py

```diff
@@ -216,13 +235,16 @@ def __init__(self, dtype):
     def encode(self,
                data: mx.sym.Symbol,
                data_length: Optional[mx.sym.Symbol],
-               seq_len: int) -> Tuple[mx.sym.Symbol, mx.sym.Symbol, int]:
+               seq_len: int,
+               att_dict: Optional[Dict[str, mx.sym.Symbol]] = None) -> Tuple[mx.sym.Symbol, mx.sym.Symbol, int]:
         """
         Encodes data given sequence lengths of individual examples and maximum sequence length.
 
         :param data: Input data.
         :param data_length: Vector with sequence lengths.
         :param seq_len: Maximum sequence length.
+        :param att_dict: An optional dictionary of attention matrices used for visualization.
+            Each matrix must be of size (batch_size, source_length, source_length).
```

Comment:
[nit] indentation of line 247

Reply:
what's the correct way of indenting multiple line doc strings? Should it be aligned with ":param" or with "An"?

Reply:
I think we have aligned it with 'An' in the past. But to be honest I dont know what the right way is :)